src/NServiceBus.Core/Extensibility/OutgoingPipelineExtensionState.cs

```diff
@@ -0,0 +1,30 @@
+namespace NServiceBus.Extensibility
+{
+    using System;
+
+    /// <summary>
+    /// Marks a given piece of state as an extension that needs to be handled before the message body is serialized.
+    /// </summary>
+    public abstract class OutgoingPipelineExtensionState
```

Comment:
Could this be moved to the delivery constraints perhaps? Or are there other use cases?